### PR TITLE
feature: editable opacity on polygons

### PR DESCRIPTION
**Pull request by FelixAbrahamsson**

### CODE REVIEW

File: src/core/drawPolygon.js [hunk 1]
```diff
@@ -76,7 +76,8 @@ const drawPolygon = (canvas, state) => (polygon, index) => {
         ) {
           const [startCoordinate] = coordinates;
           context.beginPath();
-          context.fillStyle = `${polygon.color}22`; // eslint-disable-line no-param-reassign
+          const opacity = Math.round(polygon.opacity * 255).toString(16);
```
Comment (fippli) [inline src/core/drawPolygon.js:79]: does this work? I think the opacity here assumes hex e.g. `#ffaa00[opacity]`

**Comment by fippli** [comment]
the opacity should probably be set before we even try to render so the render function just renders the color and the opacity is on the box. i.e. the box color is fixed and not adjustable like this.

**Comment by FelixAbrahamsson** [comment]
> the opacity should probably be set before we even try to render so the render function just renders the color and the opacity is on the box. i.e. the box color is fixed and not adjustable like this.

In my use case I want to be able to click on a polygon and have it change color and opacity. That would only be possible with this solution, no?

**Comment by fippli** [comment]
We can do it here now it's ok. I think this change should be done before we render and not in the render function. So you change the box color then send it to the renderer like
```
// pseudo code
changeBoxColor(boxes).map(render)
```
or whatever

instead of 
```
// pseudo code
boxes.render(b => {
  color = new color
  paintBox()
})
```

**Comment by fippli** [comment]
anyway, the only important thing here is that color
```
#ff0000
```
renders correctly. Will it?
seems weird to me that 
```
#ff00000.123
```
should be a valid color format?

**Comment by FelixAbrahamsson** [comment]
> anyway, the only important thing here is that color
> 
> ```
> #ff0000
> ```
> 
> renders correctly. Will it? seems weird to me that
> 
> ```
> #ff00000.123
> ```
> 
> should be a valid color format?

If opacity is 0.135 it renders as `#ff000022`, where `"22"` is `0.135` in hexadecimal :) It's converted on line 79. So any color renders just fine.

**Comment by fippli** [comment]
ah! sorry, I was a bit quick and didn't pay attention to line 79 that much

**Review by fippli — APPROVED** [review]

**Pull request merged**